add krew and ketall for kubectl in ops-toolbelt image

neo-liang-sap opened the pull request:
**What this PR does / why we need it**:

This PR is to add plugin manager (krew) for for kubectl and install the plugin (ketall) to kubectl in the ops-toolbelt image.
After this PR you may use krew and ketall like these
`kubectl plugin list` 
`kubectl  krew list`
`kubectl get-all` <- to get all resource in some namespace.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

for more information please refer to:
https://github.com/kubernetes-sigs/krew
https://github.com/corneliusweig/ketall

tested locally like this (after `docker build` and `docker run`)
![image](https://user-images.githubusercontent.com/42594392/75736814-25fbcf00-5d39-11ea-9e84-dd477759be36.png)

**Release note**:
```improvement user
`krew` is now available so that you can easily add additional `kubectl` plugins with `kubectl krew install {plugin_name}`
``` 
```improvement user
`kubectl` plugin `ketall` is now available. Run `kubectl get-all` to show really all kubernetes resources
```

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/gardener/ops-toolbelt?pullRequest=17) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/gardener/ops-toolbelt?pullRequest=17) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/gardener/ops-toolbelt?pullRequest=17) it.</sub>

petersutter commented:
Hi @neo-liang-sap , thank you for the PR, I really like it.

- @plkokanov we should add this also to the ghelp info. Can you point @neo-liang-sap in the right direction what needs to be changed so that it appears in the table? Or at least we should add it to the kubectl description. WDYT?
- @neo-liang-sap can you also add a release note so that users are informed? The pull request template contains a section with a template for release notes. If you need help you can contact me.

Release notes example:
``` 
`krew` is now available so that you can easily add additional `kubectl` plugins with `kubectl krew install {plugin_name}`
```
``` 
`kubectl` plugin `ketall` is now available. Run `kubectl get-all` to show really all kubernetes resources
```

petersutter commented:
cc @mandelsoft

neo-liang-sap commented:
Hello @petersutter :

Thanks for your kind comments!
i've edited PR comments to delete internal info and update `release notes` according to template, do let me know if needs more editing.

For the `ghelp info` do let me know where to edit and i will update this PR accordingly.

Thanks!
-Neo

petersutter commented:
Hi @neo-liang-sap ,

if you want to update the kubectl text with the info about the plugins I guess this is the right place: https://github.com/gardener/ops-toolbelt/blob/e33a8ad06990431aec0ea9e80d985ddae83b3a2f/dockerfile-configs/common-components.yaml#L49

But I'm wondering why https://github.com/gardener/ops-toolbelt/pull/17/files#diff-65f08d4ff14db517cebec02f6453205eR123 and https://github.com/gardener/ops-toolbelt/pull/17/files#diff-65f08d4ff14db517cebec02f6453205eR128 is not appearing when using ghelp 🤔 @plkokanov can help here. 

Another thing regarding the release notes: Please split them up into two seperate code blocks. The reason is that the first line of a release note block will be a top level bullet point and the second line will be a second level bullet point. So it would look like the following:
- `krew` is now available so that you can easily add additional `kubectl` plugins with `kubectl krew install {plugin_name}`
  - `kubectl` plugin `ketall` is now available. Run `kubectl get-all` to show really all kubernetes resources

neo-liang-sap commented:
Hi @petersutter:
Thanks for the comments and i've updated release notes.

For why `info` is not included in `ghelp` - the `ghelp` info is generated in https://github.com/gardener/ops-toolbelt/blob/master/generator/generate-dockerfile.py#L33 based on `command_list` and `command_list` is generated in https://github.com/gardener/ops-toolbelt/blob/0b59779ce14cf7686c7a4463aafd733f3d5513f0/generator/lib/commands.py#L138

only command in type `apt` `pip` `downloaded_commands` will be included, the two command i added are in `bash` so it's not included.

Do you think it would be ok to add `kubectl get-all`  and `kubectl  krew` info in  https://github.com/gardener/ops-toolbelt/blob/e33a8ad06990431aec0ea9e80d985ddae83b3a2f/dockerfile-configs/common-components.yaml#L49 as subcommand info of `kubectl`

Or i will raise another issue to add `bash` command support in types of `command_list`?

Thanks!
-Neo

petersutter commented:
Ah ok, thanks for the info.

From my point of view it's sufficient to add it as subcommand info of `kubectl`

Regarding the release notes, you need to surround the release note with a code block (see also pr template as reference) so that it get's picked up
<img width="824" alt="Screenshot 2020-03-03 at 15 53 13" src="https://user-images.githubusercontent.com/5526658/75787585-31fe8580-5d67-11ea-94ff-7e1df5dff957.png">

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/gardener/ops-toolbelt?pullRequest=17) <br/>All committers have signed the CLA.

neo-liang-sap commented:
Hi @petersutter 
Thanks, i've edit release notes according to your screenshot.

For the long one line `info` in `kubectl` - i didn't figure our one way to put it in multi-line, `\n\` seems doesn't help and /hacks/ghelp will report some parsing error - i suspect there might be some bug in parsing the `info` but didn't look into deep

Thanks!
-Neo

petersutter commented on dockerfile-configs/common-components.yaml:
```diff
@@ -46,7 +46,7 @@
     version: v1.17.1
     from: https://storage.googleapis.com/kubernetes-release/release/{version}/bin/linux/amd64/kubectl
-    info: command line tool for controlling Kubernetes clusters
+    info: command line tool for controlling Kubernetes clusters. "kubectl krew install plunin_name" to install plugin. "kubectl get-all -n namespace" to list all resource in namespaces.
```
```suggestion
    info: command line tool for controlling Kubernetes clusters. "kubectl krew install plunin_name" to install plugin. "kubectl get-all -n namespace" to list all resources in namespaces.
```

neo-liang-sap replied:
thanks! i've fixed this.

neo-liang-sap commented:
Thanks for approving! @petersutter @plkokanov 
I don't have write access to this repo so i can't merge this PR - could you please merge this PR? or grant me with collaborator role :)
